packages/mds-types/transformers/0_4_1_to_1_0_0/index.ts

```diff
@@ -0,0 +1,117 @@
+  [P in VEHICLE_EVENT_0_4_1 | TRANSFORMER_VEHICLE_EVENT]: {
+    [Q in VEHICLE_REASON_0_4_1 | TRANSFORMER_EVENT_TYPE_REASON]: {
+      event_type: VEHICLE_EVENT
+      vehicle_state: VEHICLE_STATE
+    }
+  }
+} = {
+  /* `agency_drop_off` included for the sake of completeness. No provider will ever submit such
+   * an event type through the agency API.
+   */
+  agency_drop_off: { no_event_type_reason: { event_type: 'agency_drop_off', vehicle_state: 'available' } },
+  agency_pick_up: { no_event_type_reason: { event_type: 'agency_pick_up', vehicle_state: 'removed' } },
+  cancel_reservation: { no_event_type_reason: { event_type: 'reservation_cancel', vehicle_state: 'available' } },
+  deregister: {
+    decommissioned: { event_type: 'decommissioned', vehicle_state: 'removed' },
+    missing: { event_type: 'missing', vehicle_state: 'removed' }
+  },
+  provider_pick_up: {
+    rebalance: { event_type: 'rebalance_pick_up', vehicle_state: 'removed' },
+    maintenance: { event_type: 'maintenance_pick_up', vehicle_state: 'removed' },
+    charge: { event_type: 'maintenance_pick_up', vehicle_state: 'removed' },
+    compliance: { event_type: 'compliance_pick_up', vehicle_state: 'removed' }
+  },
+  provider_drop_off: {
+    no_event_type_reason: { event_type: 'provider_drop_off', vehicle_state: 'available' }
+  },
+  reserve: {
+    no_event_type_reason: { event_type: 'reservation_start', vehicle_state: 'reserved' }
+  },
+  service_start: {
+    no_event_type_reason: { event_type: 'on_hours', vehicle_state: 'available' }
+  },
+  service_end: {
+    low_battery: { event_type: 'battery_low', vehicle_state: 'non_operational' },
+    maintenance: { event_type: 'maintenance', vehicle_state: 'non_operational' },
+    compliance: { event_type: 'compliance_pick_up', vehicle_state: 'non_operational' },
+    off_hours: { event_type: 'off_hours', vehicle_state: 'non_operational' }
+  },
+  trip_end: {
+    no_event_type_reason: { event_type: 'trip_end', vehicle_state: 'available' }
+  },
+  trip_enter: {
+    no_event_type_reason: { event_type: 'trip_enter_jurisdiction', vehicle_state: 'on_trip' }
+  },
+  trip_leave: {
+    no_event_type_reason: { event_type: 'trip_leave_jurisdiction', vehicle_state: 'elsewhere' }
+  },
+  trip_start: {
+    no_event_type_reason: { event_type: 'trip_start', vehicle_state: 'on_trip' }
+  },
+  /* We don't actually accept events with this event_type in Agency, but it's part of
+   * the VEHICLE_EVENT_0_4_1 type, and Omit doesn't seem to work on types that
+   * are string arrays, so it has to be here.
+   */
+  register: {
+    no_event_type_reason: { event_type: 'unspecified', vehicle_state: 'unknown' }
+  },
+  /* This event_type exists only to ensure backconversions and should not be present in any
+   * real events submitted via 0.4.1.
+   */
+  no_backconversion_available: {
+    no_event_type_reason: { event_type: 'unspecified', vehicle_state: 'unknown' }
+  }
+}
+
+function map_v_0_4_1_fields_to_v_1_0_0_fields(
```

Comment:
Unless this is going to live in a file that is explicitly for VehicleEvents, I think this needs more explicit naming about _what_ fields it is mapping.

Reply:
agree